internal/monitors/telegraf/common/accumulator/accumulator.go

```diff
@@ -57,9 +76,69 @@ func (ac *Accumulator) AddError(err error) {
 	ac.emit.AddError(err)
 }
 
+// AddMetric - adds a metric and will use the configured
+func (ac *Accumulator) AddMetric(m telegraf.Metric) {
+	m.Accept() // TOOD: mark telegraf tracking metrics correctly in emitter
+	ac.emit.AddMetric(m)
+}
+
+// AddMetrics - is a convenience function for adding multiple metrics to the
+// telegraf accumulator
+func (ac *Accumulator) AddMetrics(ms []telegraf.Metric) {
+	for _, m := range ms {
+		ac.AddMetric(m)
+	}
+}
+
+// WithTracking - returns an accumulator that supports tracking metrics
+func (ac *Accumulator) WithTracking(max int) telegraf.TrackingAccumulator {
+	return &TrackingAccumulator{
+		Accumulator: ac,
+		done:        make(chan telegraf.DeliveryInfo, max),
+	}
+}
+
 // NewAccumulator returns a pointer to an Accumulator
 func NewAccumulator(e emitter.Emitter) *Accumulator {
 	return &Accumulator{
 		emit: e,
 	}
 }
+
+// TrackingAccumulator will signal after a metric has been processed.  It indicates
+// if the metric was accepted, rejected, or dropped
+type TrackingAccumulator struct {
+	*Accumulator
+	done chan telegraf.DeliveryInfo
+}
+
+// AddTrackingMetric - Adds a metric to track on the the tracking accumulator
+func (ta *TrackingAccumulator) AddTrackingMetric(m telegraf.Metric) telegraf.TrackingID {
+	// TODO: maybe we'll care about these at some point,
+	// but for now dont worry about tracking metrics
+	met, id := metric.WithTracking(m, ta.deliveryCallback)
+	ta.AddMetric(met)
+	return id
+}
+
+// AddTrackingMetricGroup - Adds a group of metrics on the tracking accumulator
+func (ta *TrackingAccumulator) AddTrackingMetricGroup(group []telegraf.Metric) telegraf.TrackingID {
+	// TODO: maybe we'll care about these at some point,
+	// but for now dont worry about tracking metrics
+	mets, id := metric.WithGroupTracking(group, ta.deliveryCallback)
+	ta.AddMetrics(mets)
+	return id
+}
+
+// Delivered -
+func (ta *TrackingAccumulator) Delivered() <-chan telegraf.DeliveryInfo {
+	return ta.done
+}
+
+func (ta *TrackingAccumulator) deliveryCallback(i telegraf.DeliveryInfo) {
+	select {
+	case ta.done <- i:
+	default:
+		ta.Accumulator.emit.AddDebug(fmt.Sprintf("unable to return delivery Status '%t' for ID '%v' channel is full", i.Delivered(), i.ID()))
```

Comment:
If you make `AddDebug`'s signature `(msg string, args ...interface{})` then you can just do the `Sprintf` in that method and use it like the normal `logrus.Debugf` function.

Reply:
ok will update